packages/react-aria-components/docs/GridList.mdx

```diff
@@ -605,6 +615,953 @@ function AsyncList() {
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+.react-aria-Item {
+  &[draggable] {
+    padding-left: 4px;
+  }
+
+  &[data-dragging] {
+    opacity: 0.6;
+  }
+
+  [slot=drag] {
+    all: unset;
+    width: 15px;
+    text-align: center;
+
+    &[data-focus-visible] {
+      border-radius: 4px;
+      box-shadow: 0 0 0 2px var(--highlight-background);
+    }
+  }
+
+  &[aria-selected=true] {
+    [slot=drag][data-focus-visible] {
+      box-shadow: 0 0 0 2px var(--highlight-foreground);
+    }
+  }
+}
+
+.react-aria-DropIndicator[data-drop-target] {
+  outline: 1px solid var(--highlight-background);
+}
+```
+
+</details>
+
+### Custom drag preview
+
+By default, the drag preview shown under the user's pointer or finger is a copy of the original element that started the drag. A custom preview can be rendered by implementing the `renderDragPreview` function, passed to `useDragAndDrop`. This receives the dragged data that was returned by `getItems`, and returns a rendered preview for those items.
+
+This example renders a custom drag preview which shows the number of items being dragged.
+
+```tsx example
+import {useListData} from 'react-stately';
+import {useDragAndDrop} from 'react-aria-components';
+
+function Example() {
+  let {dragAndDropHooks} = useDragAndDrop({
+    // ...
+    ///- begin collapse -///
+    getItems: (keys) => [...keys].map(key => ({'text/plain': list.getItem(key).name})),
+    onReorder(e) {
+      if (e.target.dropPosition === 'before') {
+        list.moveBefore(e.target.key, e.keys);
+      } else if (e.target.dropPosition === 'after') {
+        list.moveAfter(e.target.key, e.keys);
+      }
+    },
+    ///- end collapse -///
+    ///- begin highlight -///
+    renderDragPreview(items) {
+      return (
+        <div className="drag-preview">
+          {items[0]['text/plain']}
+          <span className="badge">{items.length}</span>
+        </div>
+      );
+    }
+    ///- end highlight -///
+  });
+
+  ///- begin collapse -///
+  let list = useListData({
+    initialItems: [
+      {id: 1, name: 'Adobe Photoshop'},
+      {id: 2, name: 'Adobe XD'},
+      {id: 3, name: 'Adobe Dreamweaver'},
+      {id: 4, name: 'Adobe InDesign'},
+      {id: 5, name: 'Adobe Connect'}
+    ]
+  });
+
+  return (
+    <MyGridList aria-label="Reorderable list" selectionMode="multiple" items={list.items} dragAndDropHooks={dragAndDropHooks}>
+      {item => <MyItem>{item.name}</MyItem>}
+    </MyGridList>
+  );
+  ///- end collapse -///
+  // ...
+}
+```
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+.drag-preview {
+  width: 150px;
+  padding: 4px 8px;
+  display: flex;
+  justify-content: space-between;
+  background: slateblue;
+  color: white;
+  border-radius: 4px;
+
+  .badge {
+    background: white;
+    color: slateblue;
+    padding: 0 8px;
+    border-radius: 4px;
+  }
+}
+```
+
+</details>
+
+### Drag data
```

Comment:
Nit: the dragged items `data-dragging="true"` doesn't update to false/undefined when the drop is completed. Double checked locally and the `isDragging` from the draggableCollectionState is properly returning false on drag end so maybe something else isn't updating/rerendering properly. This doesn't seem to happen for the "drop on collection" or the "Dropping between items" example

EDIT: same thing happens for ListBox/Table as well

Reply:
in what browser? it seems to be working for me in Chrome, Firefox, and Safari.

Reply:
Ah my bad, I should've been more specific. This is specifically with keyboard DnD in Chrome, Firefox, and Safari. Here are two of the examples that exhibit the issue: https://reactspectrum.blob.core.windows.net/reactspectrum/34bfb98d451290f7717eee67b39f508f276954d6/docs/react-aria/GridList.html#text
https://reactspectrum.blob.core.windows.net/reactspectrum/34bfb98d451290f7717eee67b39f508f276954d6/docs/react-aria/GridList.html#dropping-on-items

Reply:
Ah I see this too. I think it's related to the `dragState.isDragging` checking a ref, which is updated after setting state rather than before. https://github.com/adobe/react-spectrum/blob/1e74a5a0d4981212f1f5aef145f10726d5edfc19/packages/%40react-stately/dnd/src/useDraggableCollectionState.ts#L134

Will try to change this in a followup and see if anything breaks.